rwengine/src/ai/CharacterController.cpp

```diff
@@ -252,12 +252,12 @@ bool Activities::EnterVehicle::update(CharacterObject *character,
                     character->getCurrentState().armour = 100.f;
                 } else if (vehicle->getVehicle()->vehiclename_ == "AMBULAN") {
                     // Receive up to 25 HPs from Ambulance
-                    if (character->getCurrentState().health < 100.f) {
-                        if (character->getCurrentState().health > 75.f) {
-                            character->getCurrentState().health = 100.f;
-                        } else {
-                            character->getCurrentState().health += 25.f;
-                        }
+                    float health = character->getCurrentState().health;
+                    // todo: Replace by correct fullHealth once implemented
+                    float fullHealth = 100.f;
+                    if (health < fullHealth) {
+                        character->getCurrentState().health =
+                            (health >= 80.f) ? fullHealth : health + 20.f;
```

Comment:
That I don't know, don't have a 100% finished savegame. Perhaps should get one and test it.

Prostitutes give up to extra 25 health above maximum of 100 in GTA-III. If the 100% game completion raises the fullHealth value, perhaps it affects the math and lets you have more than 200 HPs. Although I it's related to Vice City and partially to SA and LCS/VCS, not GTA-III.